src/utils/loginPrompt/index.ts

```diff
@@ -0,0 +1,167 @@
+enum domIDs {
+  styles = 'sasjsAdapterStyles',
+  overlay = 'sasjsAdapterLoginPromptBG',
+  dialog = 'sasjsAdapterLoginPrompt'
+}
+
+export const openLoginPrompt = (): Promise<boolean> => {
+  return new Promise(async (resolve) => {
+    const style = document.createElement('style')
+    style.id = domIDs.styles
+    style.innerText = cssContent
+
+    const loginPromptBG = document.createElement('div')
+    loginPromptBG.id = domIDs.overlay
+    loginPromptBG.classList.add('popUpBG')
+
+    const loginPrompt = document.createElement('div')
+    loginPrompt.id = domIDs.dialog
+    loginPrompt.classList.add('popUp')
+
+    const title = document.createElement('h1')
+    title.innerText = 'Session Expired!'
+    loginPrompt.appendChild(title)
+
+    const descHolder = document.createElement('div')
+    const desc = document.createElement('span')
+    desc.innerText = 'You need to relogin, click OK to login.'
+    descHolder.appendChild(desc)
+    loginPrompt.appendChild(descHolder)
+
+    const buttonCancel = document.createElement('button')
+    buttonCancel.classList.add('cancel')
+    buttonCancel.innerText = 'Cancel'
+    buttonCancel.onclick = () => {
+      closeLoginPrompt()
+      resolve(false)
+    }
+    loginPrompt.appendChild(buttonCancel)
+
+    const buttonOk = document.createElement('button')
+    buttonOk.classList.add('confirm')
+    buttonOk.innerText = 'Ok'
+    buttonOk.onclick = () => {
+      closeLoginPrompt()
+      resolve(true)
+    }
+    loginPrompt.appendChild(buttonOk)
+
+    document.body.style.overflow = 'hidden'
+
+    document.body.appendChild(style)
+    document.body.appendChild(loginPromptBG)
+    document.body.appendChild(loginPrompt)
+  })
+}
+const closeLoginPrompt = () => {
+  Object.keys(domIDs).forEach((id) => {
+    const elem = document.getElementById(id)
+    elem?.parentNode?.removeChild(elem)
+  })
+
+  document.body.style.overflow = 'auto'
+}
+
+const cssContent = `
+.popUp {
```

Comment:
Have we checked this at a few different resolutions? I see you've mostly used percentages for widths and heights so we should be okay, but I just want to be completely sure.